lib/matplotlib/mlab.py

```diff
@@ -3874,23 +3874,26 @@ def contiguous_regions(mask):
     """
     return a list of (ind0, ind1) such that mask[ind0:ind1].all() is
     True and we cover all such regions
-
-    TODO: this is a pure python implementation which probably has a much
-    faster numpy impl
     """
+    mask = np.asarray(mask, dtype=bool)
+
+    if not mask.size:
+        return []
+
+    # Find the indices of region changes, and correct offset
+    idx, = np.nonzero(mask[:-1] != mask[1:])
+    idx += 1
 
-    in_region = None
-    boundaries = []
-    for i, val in enumerate(mask):
-        if in_region is None and val:
-            in_region = i
-        elif in_region is not None and not val:
-            boundaries.append((in_region, i))
-            in_region = None
+    # Add first and/or last index if needed
```

Comment:
While we are polishing this bikeshed: it's probably most efficient to convert idx to a list at this point.  Then the prepending and/or appending can be done with a total of 4 lines, as in my example.  These operations are very fast with python lists--probably faster than np.concatenate.  I suspect the final indexing and zip operation would also be at least as fast if idx is already a list.